sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -356,26 +380,35 @@ object InMemoryFileIndex extends Logging {
           bulkListLeafFiles(
             dirs.map(_.getPath),
             hadoopConf,
-            filter,
+            filters,
             session,
-            areRootPaths = false
+            areRootPaths = false,
+            parameters = parameters
           ).flatMap(_._2)
         case _ =>
           dirs.flatMap { dir =>
             listLeafFiles(
               dir.getPath,
               hadoopConf,
-              filter,
+              filters,
               sessionOpt,
               ignoreMissingFiles = ignoreMissingFiles,
               ignoreLocality = ignoreLocality,
-              isRootPath = false)
+              isRootPath = false,
+              parameters = parameters)
           }
       }
+      val fileFilters = (filters ++ FileModifiedDateOption.accept(
+        parameters,
+        sessionOpt.get,
```

Comment:
Hi, @cchighman . This is the root cause of the failure. Could you fix and make the tests pass?
```
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:529)
	at scala.None$.get(Option.scala:527)
	at org.apache.spark.sql.execution.datasources.InMemoryFileIndex$.listLeafFiles(InMemoryFileIndex.scala:403)
```

Reply:
Yes, thank you.

Reply:
You can choose some tests in the following report in order to verify your fix. Please push your commit after passing those UT locally.
- https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124247/testReport/